### ISSUE TITLE
Unable to emit from outside (socket.io-parser is encoding message as undefined)

### ISSUE THREAD

**Issue by deepanshumehndiratta** [socketio/socket.io#2402]
socket.io-parser is encoding message as undefined

Debug log:

Server:

```
socket.io-parser encoding packet {"nsp":"/","type":2,"data":["tweet","test"]} +33s
socket.io-parser encoded {"nsp":"/","type":2,"data":["tweet","test"]} as 2["tweet","test"] +0ms
socket.io:client writing packet ["2[\"tweet\",\"test\"]"] +10s
socket.io-parser encoding packet ["2[\"tweet\",\"test\"]"] +0ms
socket.io-parser encoded ["2[\"tweet\",\"test\"]"] as undefined +0m
engine:socket sending packet "message" (undefined) +0ms
engine:socket flushing buffer to transport +0ms
engine:ws writing "4undefined" +0ms
```

Browser:

```
engine.io-client:socket socket receive: type "message", data "undefined" +3s
```

Python code:

```
io = Emitter({'host': 'localhost', 'port':6379})
io.In('stream').Emit('tweet', 'test')
```

Version info:

socket.io-emitter (Python): 0.1.3
Node: v0.10.37
socket.io: 1.4.28
socket.io-redis: 1.4.28
socket.io-parser: 1.4.28

**Comment by deepanshumehndiratta** [comment]
Reverting to 1.3.5 as suggested in https://github.com/socketio/socket.io/issues/2378 solved it.

**Comment by iolloyd** [comment]
great work @mfornasa

**Comment by jaykhawaja** [comment]
Had a similar issue with socket.io (1.7.2). The socket.io client and server successfully connects but does not io.to('room).emit('message'); Upon debugging I received a similar server log as deepanshumehndiratta. 
Reverting to 1.3.7 works for me. 

Any fixes for 1.7.2?
